source/common/conn_pool/conn_pool_base.h

```diff
@@ -192,8 +193,13 @@ class ConnPoolImplBase : protected Logger::Loggable<Logger::Id::pool> {
 
   void onConnectionEvent(ActiveClient& client, absl::string_view failure_reason,
                          Network::ConnectionEvent event);
-  // See if the drain process has started and/or completed.
-  void checkForDrained();
+
+  // Returns true if the pool is idle.
+  bool isIdleImpl() const;
+
+  // See if the pool has gone idle. If we're draining, this will also close idle connections.
+  void checkForIdle();
```

Comment:
Up to you, but for clarity I might go for the super verbose: `checkForIdleAndCloseIfDraining()`